### PR TITLE
Update compiler options for TS 2.1

### PR DESCRIPTION
**Pull request by mhegazy**

### CODE REVIEW

File: pages/Compiler Options.md [hunk 1]
```diff
@@ -6,9 +6,10 @@ Option                                         | Type      | Default
 `--allowSyntheticDefaultImports`               | `boolean` | `module === "system"`          | Allow default imports from modules with no default export. This does not affect code emit, just typechecking.
 `--allowUnreachableCode`                       | `boolean` | `false`                        | Do not report errors on unreachable code.
 `--allowUnusedLabels`                          | `boolean` | `false`                        | Do not report errors on unused labels.
+`--alwaysStrict     `                          | `boolean` | `false`                        | Parse in strict mode and emit `"use strict"` for each source file
```
Comment (sandersn) [inline pages/Compiler Options.md]: Too many spaces between the t and the backtick.

File: pages/Compiler Options.md [hunk 2]
```diff
@@ -17,29 +18,31 @@ Option                                         | Type      | Default
 `--experimentalDecorators`<sup>[1]</sup>       | `boolean` | `false`                        | Enables experimental support for ES7 decorators.
 `--forceConsistentCasingInFileNames`           | `boolean` | `false`                        | Disallow inconsistently-cased references to the same file.
 `--help`<br/>`-h`                              |           |                                | Print help message.
+`--importHelpers  `                            | `string`  |                                | Import emit helpers (e.g. `__extends`, `__rest`, etc..) from [`tslib`](https://www.npmjs.com/package/tslib)
```
Comment (sandersn) [inline pages/Compiler Options.md]: too many spaces here too. Or is this intentional for formatting?

File: pages/tsconfig.json.md [hunk 3]
```diff
@@ -133,6 +133,56 @@ Specify `"types": []` to disable automatic inclusion of `@types` packages.
 Keep in mind that automatic inclusion is only important if you're using files with global declarations (as opposed to files declared as modules).
 If you use an `import "foo"` statement, for instance, TypeScript may still look through `node_modules` & `node_modules/@types` folders to find the `foo` package.
 
+## Configuration inheritance with `extends`
+
+A `tsconfig.josn` file can inherit comfigurations from anohter file using the `extends` propeorty.
```
Comment (sandersn) [inline pages/tsconfig.json.md]: typo: josn -> json

Comment (sandersn) [inline pages/tsconfig.json.md]: typo: comfiguration -> configuration

Comment (sandersn) [inline pages/tsconfig.json.md]: typo: anohter -> another

Comment (sandersn) [inline pages/tsconfig.json.md]: typo: propeorty -> property

File: pages/tsconfig.json.md [hunk 4]
```diff
@@ -133,6 +133,56 @@ Specify `"types": []` to disable automatic inclusion of `@types` packages.
...
+
+The `extends` is a top-level propoerty in `tsconfig.json` (alongside `compilerOptions`, `files`, `include`, and `exclude`).
```
Comment (sandersn) [inline pages/tsconfig.json.md]: propoerty again

File: pages/Compiler Options.md [hunk 5]
```diff
@@ -17,29 +18,31 @@ Option                                         | Type      | Default
...
+`--jsxFactory`                                 | `string`  | `"React.createElement"`        | Specify the JSX factory function to use when targeting react JSX emit, e.g. `React.createElement` or `h`.
+`--lib`                                        | `string[]`|                                | List of library files to be included in the compilation.<br/>Possible values are:  <br/>► `ES5` <br/>► `ES6` <br/>► `es2015` <br/>► `es7` <br/>► `es2016` <br/>► `es2017` <br/>► `dom` <br/>► `webworker` <br/>► `scripthost` <br/>► `es2015.core` <br/>► `es2015.collection` <br/>► `es2015.generator` <br/>► `es2015.iterable` <br/>► `es2015.promise` <br/>► `es2015.proxy` <br/>► `es2015.reflect` <br/>► `es2015.symbol` <br/>► `es2015.symbol.wellknown` <br/>► `es2016.array.include` <br/>► `es2017.object` <br/>► `es2017.sharedmemory` <br/><br/> Note: If `--lib` is not specified a default library is injected. The default library injected is:  <br/> ► For `--target ES5`: `dom,ES5,scripthost`<br/>  ► For `--target ES6`: `dom,ES6,dom.iterable,scripthost`
```
Comment (sandersn) [inline pages/Compiler Options.md]: why uppercase only ES5 and ES6 and nothing else?

File: pages/Compiler Options.md (hunk 1 above)
Comment (mhegazy) [inline pages/Compiler Options.md]: no. just one.

Comment (mhegazy) [inline pages/Compiler Options.md]: ah.. i see it now. fixed

File: pages/Compiler Options.md (hunk 2 above)
Comment (mhegazy) [inline pages/Compiler Options.md]: done.

**Pull request merged**